src/bundler/bundle_v2.zig

```diff
@@ -2309,27 +2310,72 @@ pub const ParseTask = struct {
         };
     }
 
-    pub const runtime = ParseTask{
-        .ctx = undefined,
-        .path = Fs.Path.initWithNamespace("runtime", "bun:runtime"),
-        .side_effects = _resolver.SideEffects.no_side_effects__pure_data,
-        .jsx = options.JSX.Pragma{
-            .parse = false,
-            // .supports_react_refresh = false,
-        },
-        .contents_or_fd = .{
-            .contents = @as(string, @embedFile("../runtime.js")),
-        },
-        .source_index = Index.runtime,
-        .loader = Loader.js,
-    };
-    pub const runtime_source = Logger.Source{
-        .path = ParseTask.runtime.path,
-        .key_path = ParseTask.runtime.path,
-        .contents = ParseTask.runtime.contents_or_fd.contents,
-        .index = Index.runtime,
+    const RuntimeSource = struct {
+        parse_task: ParseTask,
+        source: Logger.Source,
     };
 
+    fn getRuntimeSourceComptime(comptime target: options.Target) RuntimeSource {
+        const code = @embedFile("../runtime.js") ++ switch (target) {
+            .bun, .bun_macro =>
+            \\export var __require = /* @__PURE__ */ import.meta.require;
+            \\
+            ,
+            .node =>
+            \\import { createRequire } from "node:module";
+            \\export var __require = /* @__PURE__ */ createRequire(import.meta.url);
+            ,
+            // Copied from esbuild's runtime.go:
```

Comment:
Do we reference ESBuilds MIT license in say a thirdparty directory? It's MIT we should include it somewhere.

Reply:
there is a very small note here. https://bun.sh/docs/project/licensing

the entire file is reimplemented off of esbuild, perhaps this could be noted.